plugins/inputs/filecount/filecount.go

```diff
@@ -127,6 +128,9 @@ func absDuration(x time.Duration) time.Duration {
 
 func count(basedir string, recursive bool, countFn countFunc) error {
 	walkFn := func(path string, file os.FileInfo, err error) error {
+		if file == nil {
+			return fmt.Errorf("Nil file found at %q", path)
```

Comment:
Check the err argument and return it, I think it should have a better error string.